File: RestaurantApp/app/src/main/java/com/example/restaurantapp/MainActivity.java
package com.example.restaurantapp;

import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    CardView startersCard;
    CardView mainsCard;
    CardView dessertsCard;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        startersCard = findViewById(R.id.card_view_starters);
        mainsCard = findViewById(R.id.card_view_mains);
        dessertsCard = findViewById(R.id.card_view_desserts);

        dessertsCard.setOnClickListener(view -> {

            Intent dessertsActivityIntent = new Intent(MainActivity.this, DessertsActivity.class);

            startActivity(dessertsActivityIntent);
        });

        startersCard.setOnClickListener(view -> {

            Intent startersActivityIntent = new Intent(MainActivity.this, StartersActivity.class);

            startActivity(startersActivityIntent);
        });

       mainsCard.setOnClickListener(view -> {

           Intent mainCoursesActivityIntent = new Intent(MainActivity.this, MainCoursesActivity.class);

           startActivity(mainCoursesActivityIntent);

       });

        TextView emailTextView = findViewById(R.id.text_view_email_address);
        emailTextView.setOnClickListener(view -> {
            Intent launchEmailAppIntent = new Intent(Intent.ACTION_SENDTO);
            launchEmailAppIntent.setData(Uri.parse("mailto: test@email.com"));
            startActivity(launchEmailAppIntent);
        });
    }
}

File: MyForm/app/src/main/java/com/example/myform/MainActivity.kt
package com.example.myform

import android.content.Intent
import android.os.Bundle
import android.widget.ArrayAdapter
import androidx.appcompat.app.AppCompatActivity
import com.example.myform.databinding.ActivityMainBinding
import com.example.myform.databinding.ActivityMainBinding.inflate

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        binding.buttonToSummary.setOnClickListener{
            onToSummaryClicked()
        }

        val spinnerValues = arrayOf("Miss", "Mrs.", "Mr.")

        val spinnerAdapter = ArrayAdapter(this, android.R.layout.simple_spinner_dropdown_item,
        spinnerValues)

        binding.spinnerTitle.adapter = spinnerAdapter
    }

    private fun onToSummaryClicked() {

        val summary = Summary(
            binding.spinnerTitle.selectedItem?.toString(),
            binding.editTextFirstName.text.toString(),
            binding.editTextLastName.text.toString()
        )

        val intent = Intent(this, UserDetails::class.java)

        intent.putExtra("Summary", summary)
        val phone = binding.editTextPhone.text.toString()
        intent.putExtra("Phone", phone)
        val email = binding.editTextEmail.text.toString()
        intent.putExtra("Email", email)

        startActivity(intent)
    }
}


